test/integration/test_job.py

```diff
@@ -161,6 +161,7 @@ def test_cancel_job_running(self, service):
             service,
             circuits=[ReferenceCircuits.bell()] * 10,
         )
+        wait_for_status(job, JobStatus.CANCELLED)
```

Comment:
I believe this addition does not really fix the problem, because it causes the test to be skipped if the `job.status() != CANCELLED`, and does not really check if the job was cancelled. See inside `wait_for_status`. We need some additional investigation.

Reply:
@merav-aharoni apologies, I now realise that the initial solution was not ideal. Furthermore, I was checking the status before calling the cancel function. 

As a possible solution, I was thinking about repeatedly querying the server to check the status of the job and waiting for the status to change to CANCELLED on the server side to provide a more accurate validation.

Reply:
@robotAstray - please see my comment in #1019 .

Reply:
@merav-aharoni thank you, I have made the changes and tested it locally. It was successful. I will push the changes.